### PR TITLE
Improve Facebook integration

### PR DESCRIPTION
**Pull request by friscoMad**
Add options to change attachment (link) caption, description, image and name

Tested in my installation, full backwards compatible.
pokemad_2017-03-20_11-05-40

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Will update Facebook wiki page after this is considered for merging 

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kvangent** [comment]
I like the changes, and think they look good, but can we get the documentation updated [here](https://github.com/kvangent/PokeAlarmWiki).

Also, what are the parameters for pictures? Is it a URL to a picture? Because if so, we can add in some defaults for the rest of the fields to improve the look of the Facebook message from the get go.

**Comment by friscoMad** [comment]
Sure, should I do a PR for that repo too?

Yup the picture is a URL we can use the same default as discord, I did it that way to avoid changing the current behaviour in old installations, the small map picture can be found more usefull for some people.
I think that if we provide a default there is no way to disable it in config (as None will use the default), that's the reason I thought it will be better to provide an example in docs over providing a default in code.

**Comment by kvangent** [comment]
Yes, go ahead and make a PR to that wiki repo.

What we did in other alarms is setting it to 'False' turns it off.

**Comment by friscoMad** [comment]
Can you point me where can I find such optional configuration?
I have little experience in python but I am unable to see anything like that in any other connector.

**Comment by kvangent** [comment]
In Telegram, there are a bunch such as Location and Venue that should be about what you need.

**Comment by friscoMad** [comment]
Updated with default image as I am not sure what could be the default values for other fields, and made a PR in doc repo with this new options.

**Comment by kvangent** [comment]
I've got some more feedback for you (and I've poked Remedy to take a look at your wiki PR):

- Message default to `status message	A wild <pkmn> has appeared!`
- Default name to `<pkmn>`
- Description default to `Available until <24h_time> (<time_left>).`

**Comment by friscoMad** [comment]
@kvangent done

**Comment by kvangent** [comment]
@friscoMad You've got some tabs in there

**Comment by RamiroWincor** [comment]
@kvangent My bad I did configure it only in one PC, updated without tabs.

**Comment by kvangent** [comment]
Remedy and I are both just swamped with non-PA stuff right now, but I should be able to finalize this after this weekend is over.

File: PokeAlarm/FacebookPage/FacebookPageAlarm.py [hunk 1]
```diff
@@ -23,15 +23,20 @@ class FacebookPageAlarm(Alarm):
...
-            'message': "A wild <pkmn> has appeared! Available until <24h_time> (<time_left>).",
-            'link': "<gmaps>"
+            'message': "A wild <pkmn> has appeared!.",
+            'picture': "https://raw.githubusercontent.com/kvangent/PokeAlarm/master/icons/<pkmn_id>.png",            
+            'link': "<gmaps>",
+            'name': "<pkmn>", 
+            'description': "Available until <24h_time> (<time_left>)"
         },
         'pokestop': {
             'message': "Someone has placed a lure on a Pokestop! Lure will expire at <24h_time> (<time_left>).",
```
Comment (kvangent) [inline PokeAlarm/FacebookPage/FacebookPageAlarm.py]: Can we get some defaults for the other alert types added? Suggestions:

pokestop:
- Message: `"Someone has placed a lure on a Pokestop!"`
- Name: `"Lured Pokestop"`
- Description: `"Lure will expire at <24h_time> (<time_left>)."`
- Caption: `None` <- Just so it is clear this is an option

Gym:
- Message: `"Team <new_team> has claimed a new gym! "`
- Name: `"It is now controlled by <new_team>."`
- Description: `"Lure will expire at <24h_time> (<time_left>)."`
- Caption: `None` <- Just so it is clear this is an option

File: PokeAlarm/FacebookPage/FacebookPageAlarm.py [hunk 2]
```diff
@@ -70,16 +75,29 @@ def startup_message(self):
     def create_alert_settings(self, settings, default):
         alert = {
             'message': settings.pop('message', default['message']),
-            'link': settings.pop('link', default['link'])
+            'link': settings.pop('link', default['link']),
+            'caption': settings.pop('caption', None),
+            'description': settings.pop('description', None),
+            'picture': settings.pop('picture', default['picture']),
+            'name': settings.pop('name', None)
```
Comment (kvangent) [inline PokeAlarm/FacebookPage/FacebookPageAlarm.py]: Make sure these all take from the default correctly.

**Comment by RamiroWincor** [comment]
@kvangent Done

**Comment by kvangent** [comment]
Looks good, as soon as the Wiki is good I'll pull them both.

**Comment by friscoMad** [comment]
I messed my identities :D 
There is one more commit that it is the picture -> image rename that I will push as soon as I test it is working.

**Comment by friscoMad** [comment]
Ok now I think it is finished, I neede to change "False" to "" otherwise I would need to check against "False" or at least cast to boolean first, null in json was another option but it is not used anywhere and people already has too much troubles with json, so I think "" is the best solution.

Docs already updated also, @kvangent please tell me if you want more changes.

File: PokeAlarm/FacebookPage/FacebookPageAlarm.py [hunk 3]
```diff
@@ -70,16 +82,29 @@ def startup_message(self):
...
         }
         reject_leftover_parameters(settings, "'Alert level in FacebookPage alarm.")
         return alert
 
     # Post Pokemon Message
     def send_alert(self, alert, info):
+        attachment = {"link": replace(alert['link'], info)}
+        if alert['caption']:
+            attachment['caption'] = replace(alert['caption'], info)
+        if alert['description']:
```
Comment (kvangent) [inline PokeAlarm/FacebookPage/FacebookPageAlarm.py]: I think I missed this before, but can any of these but caption actually be unset?

**Comment by kvangent** [comment]
@friscoMad I agree, check from empty string seems like a good solution. Can we get an addendum somewhere in the wiki docs that makes that clear?

**Pull request merged**